docs/ref/models/expressions.txt

```diff
@@ -588,22 +588,38 @@ expressions::
 
 Next, we write the method responsible for generating the SQL::
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
         sql_expressions, sql_params = [], []
         for expression in self.expressions:
             sql, params = compiler.compile(expression)
             sql_expressions.append(sql)
             sql_params.extend(params)
         self.extra['expressions'] = ','.join(sql_expressions)
-        return self.template % self.extra, sql_params
+        template = kwargs.get('template', self.template)
+        return template % self.extra, sql_params
 
-    def as_oracle(self, compiler, connection):
+    def as_oracle(self, compiler, connection, **kwargs):
         """
         Example of vendor specific handling (Oracle in this case).
         Let's make the function name lowercase.
         """
-        self.template = 'coalesce( %(expressions)s )'
-        return self.as_sql(compiler, connection)
+        return self.as_sql(compiler, connection,
+            template='coalesce( %(expressions)s )'
+        )
+
+.. note::
+
+    When working with different database-backends and multiple databases you'll
```

Comment:
I don't know if the note is necessary. I don't think anyone would use the alternate technique instead of following the documentation? We should have a ``.. versionchanged:: 1.10` note that describes how the behavior changed. You could mention the reason for that change there.